src/Response.php

```diff
@@ -608,8 +621,17 @@ protected function decodeDeflate($body)
         $zlibHeader = unpack('n', substr($body, 0, 2));
 
         if ($zlibHeader[1] % 31 === 0) {
-            return gzuncompress($body);
+            $uncompressed = gzuncompress($body);
+        } else {
+            $uncompressed = gzinflate($body);
+        }
+
+        if (false === $uncompressed) {
+            throw new Exception\RuntimeException(
+                'An error occurred during inflation'
+            );
```

Comment:
This is new behavior and requires a corresponding test.

Reply:
Tests added